addons/mail/static/src/core/web/thread_service_patch.js

```diff
@@ -40,54 +40,16 @@ patch(ThreadService.prototype, {
             thread_id: thread.id,
             thread_model: thread.model,
         });
-        thread.canPostOnReadonly = result.canPostOnReadonly;
-        thread.hasReadAccess = result.hasReadAccess;
-        thread.hasWriteAccess = result.hasWriteAccess;
-        if ("activities" in result) {
-            const existingIds = new Set();
-            for (const activity of result.activities) {
-                existingIds.add(this.store.Activity.insert(activity, { html: true }).id);
-            }
-            for (const activity of thread.activities) {
-                if (!existingIds.has(activity.id)) {
-                    this.activityService.delete(activity);
```

Comment:
There is also a cross-tab sync in activity service, we might want to move that part over to `onDelete` to keep a similar behavior.